### PR TITLE
feat: [FC-0047] Course progress and collapsing sections

### PR DESCRIPTION
**Pull request by IvanStepanok**
## Quick Preview

https://github.com/openedx/openedx-app-ios/assets/128456094/dd78b250-b349-4b58-9f11-7b820e50e0fe

## Pay attention ⚠️
❌ "COURSE_NESTED_LIST_ENABLED" removed
✅ "COURSE_DROPDOWN_NAVIGATION_ENABLED" added

Old implementations in CourseStructureNestedListView.swift and CourseStructureView.swift are removed.

Design:
https://www.figma.com/design/iZ56YMjbRMShCCDxqrqRrR/Open-edX-Mobile-App-All-Screens-v2.1?node-id=9177-44208&t=kHcQmxzck33FVrAT-0

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @IvanStepanok! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by volodymyr-chekyrta** [comment]
The arrow is pointing in the wrong direction
image

File: Course/Course/Presentation/Subviews/CustomDisclosureGroup.swift [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+                                                viewModel.router.showCourseVerticalView(
+                                                    courseID: viewModel.courseStructure?.id ?? "",
+                                                    courseName: viewModel.courseStructure?.displayName ?? "",
+                                                    title: sequential.displayName,
+                                                    chapters: course.childs,
+                                                    chapterIndex: chapterIndex,
+                                                    sequentialIndex: sequentialIndex
+                                                )
+                                            }
+                                        }, label: {
```
Comment (volodymyr-chekyrta) [inline Course/Course/Presentation/Subviews/CustomDisclosureGroup.swift:84]: Please format Button to the "each param on own line" style.
```
Button(
action: {
...
, 
label: {
...
```

**Comment by volodymyr-chekyrta** [comment]
By design, arrows should be displayed only in sections with assignments.
| Design | Device |
|----------|----------|
| image | image |

**Review by volodymyr-chekyrta — CHANGES_REQUESTED** [review]
Good overall; please address the following feedback

**Comment by IvanStepanok** [comment]
@volodymyr-chekyrta Thanks for the review! All requested changes have been made🙌

**Review by volodymyr-chekyrta — APPROVED** [review]
Looks good to me.
PR is ready for product/design review.

**Comment by sdaitzman** [comment]
Hi @IvanStepanok, this looks great overall! A couple of design questions/feedback I wanted to check on:

1. In light mode, sections should use the light theme `shade` color `#F9FAFB` as their background color. It looks like this might be missing from the video
2. In dark mode, the download icon should use the lighter `dark theme accent text` color `#879FF5` for contrast/accessibility reasons

Both of these are color-related, and the video compression makes it a bit ambiguous whether these colors are already being used, so I wanted to confirm

**Comment by IvanStepanok** [comment]
Hi @sdaitzman and thanks for paying so much attention to details!
Please, take a look at my changes. The download icon had the wrong color, but in the light theme, not in the dark☺️

Simulator Screenshot - iPhone 15 - 2024-06-04 at 23 20 50

**Comment by sdaitzman** [comment]
Thanks so much @IvanStepanok! Just to confirm, is the download icon in dark mode using [`#879FF5`](https://www.figma.com/design/iZ56YMjbRMShCCDxqrqRrR/Open-edX-Mobile-App-All-Screens-v2.1?node-id=12096-95006&t=gqaiAEFAMSBQE7Vt-11)? In that screenshot, I see a value of #5378F8, but it's possible that's just an image encoding/color space variation.

Other than that, everything looks great to me/ready to merge from a design standpoint, thanks for these updates!

Screenshot 2024-06-05 at 10 31 19 AM

**Comment by IvanStepanok** [comment]
Hi, @sdaitzman 
For some reason when you load icons and make a screenshot, they have a little difference in color measurement. I think is because of different color profiles used on real device and the simulator. I found a topic about this, but anyway, I think all we can do is just accept the rules🙌

File: Course/Course/Presentation/Outline/CourseOutlineView.swift [hunk 2]
```diff
@@ -108,21 +107,19 @@ public struct CourseOutlineView: View {
                                 ? viewModel.courseVideosStructure
                                 : viewModel.courseStructure {
                                 
-                                // MARK: - Sections
-                                if viewModel.config.uiComponents.courseNestedListEnabled {
```
Comment (rnr) [inline Course/Course/Presentation/Outline/CourseOutlineView.swift:112]: @IvanStepanok 
we had logic when `courseNestedListEnabled==true` we skip VerticalsView and go to content directly. Could you please add the same? Thank you

**Review by rnr — CHANGES_REQUESTED** [review]

File: OpenEdX/Router.swift [hunk 3]
```diff
@@ -590,13 +590,10 @@ public class Router: AuthorizationRouter,
             chapterIndex: chapterIndex,
             sequentialIndex: sequentialIndex
         )
-
-        let config = Container.shared.resolve(ConfigProtocol.self)
-        let isCourseNestedListEnabled = config?.uiComponents.courseNestedListEnabled ?? false
         
         var controllers = navigationController.viewControllers
 
-        if isCourseNestedListEnabled || currentCourseTabSelection == CourseTab.dates.rawValue {
```
Comment (rnr) [inline OpenEdX/Router.swift:599]: this condition should be present. please use new added flag here. Thank you

**Review by rnr — CHANGES_REQUESTED** [review]

File: OpenEdX/Managers/PipManager.swift [hunk 4]
```diff
@@ -114,7 +111,7 @@ public class PipManager: PipManagerProtocol {
             viewControllers.append(try await containerController(for: holder))
         }
         
-        if !isNestedListEnabled && holder.selectedCourseTab != CourseTab.dates.rawValue {
```
Comment (rnr) [inline OpenEdX/Managers/PipManager.swift:117]: the same here - please use here new added flag. Thank you

**Review by rnr — CHANGES_REQUESTED** [review]

**Review by rnr — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@IvanStepanok 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-ios&entry.867055334=https://github.com/openedx/openedx-app-ios/pull/446&entry.1484655318=https://github.com/IvanStepanok&entry.752974735=2024-05-31+13:21&entry.1917517419=2024-06-07+12:18&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**